### PR TITLE
[Bugfix] Several fixes for Programming Exercises

### LINKED ISSUE #2760: More than 10 assessments can be started and the amount of locks is not correctly displayed
### Describe the bug
Usually, a Tutor can only have 10 unfinished assessments per Exercise. This limit does not apply anymore, a Tutor can have more than 10 unfinished assessments right now. Furthermore, the locked assessments are not shown on the "Assessment Dashboard" and cannot be unlocked from there (Assessments can still be unlocked directly by the "Cancel" Option on the Assessment Screen).

#### To Reproduce
1. Start new Assessment (can be repeated more than 10 times)
2. Go to the Assessment Dashboard
3. The "Assessments locked by you:" field will show 0
4. Click on the "Assessments locked by you:" Link
5. Artemis will show "No assessments locked by you!"

#### Expected behavior
I expected that Artemis will prevent me from starting more than 10 (unfinished) assessments, as well as showing me the correct amount of currently locked assessments on the "Assessment Dashboard".

#### Screenshots
More than 10 Assessments:
MoreThan10Assessments

Assessment Dashboard showing 0 locked assessments:
AssessmentLocks
Locks

### Environment
<details><pre>
 - OS: Windows 10
 - Browser: Latest Firefox, Latest Google Chrome
 - Version: Artemis 4.9.2
</pre></details>

#### Additional context
Affected Course: 83 (PGdP - Praktikum Grundlagen der Programmierung)
Affected Exercise: 2849 (Funktionscache)

### LINKED ISSUE #2763: Tutors see buttons in the course details that they cannot / should not use
Logged in with `artemis_test_user_7`, who is only tutor for the course:

Course Management: no instructor actions:
image

Course Details: instructor actions:
image

Cklick Edit:
image

But saving does not work. Scores and Instructor Course Dashboard neither (although `stats-for-instructor-dashboard` call works, but `with-exercises-and-relevant-participations` does not). Errors only appear in the browser debug network tab, page stays empty.

### PR DESCRIPTION
**Pull request by fde312**
Fixes #2763 
Fixes #2760 

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) locally.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

This PR fixes several minor bugs found in different testing sessions (see also issues attached).

### Description

- Show result string instead of Build Failed for manual programming assessments
- Extended query for locked submission 
- Add More Feedback Request label
- Remove buttons for tutors in course and exercise details view
- Load assessor eagerly

### Steps for Testing

1. Create a Programming exercise, participate, assess and make a complaint (Care, the assessor musn't be the student)
2. Accept the complaint. After that, the original assessor should be correctly displayed (See screenshot)
3. In another submission, ask for more feedback. When the assessor clicks on the button to give more feedback on the top right corner "You have to respond to the feedback request" should be displayed (See screenshot)

1. Navigate to the course detail page and programming-exercise detail page with an student account and make sure you cannot see tutor or instructor related buttons (See screenshot)

1. Lock 10 submissions and make sure you cannot lock another one. You can see your locks in the 'Assessments locked by you' in the assessment dashboard.
1. Participate in an programming exercise with manual assessment and make sure the build fails (e.g. remove a ; in the end of a line)
2. Assess the exercise. For the student, the scores should be displayed afterwards (See screenshot)

### Test Coverage

Nothing changed.

### Screenshots

Course Detail View:
image

Exercise Detail View:
image

Locked Submission View:
image

Locked Submission Limit:
image

Build Failed result string:
image

After evaluating and accepting the complaint:

image

More feedback request:
image

### CODE REVIEW

**Review by airelawaleria — APPROVED** [review]
tested with @simonzilker @marvinchimeka @JohannesWeiss on ts1

**Review by simonzilker — DISMISSED** [review]
Tested in testing session. No issues found :)

**Review by marvinchimeka — DISMISSED** [review]

**Review by JohannesWeiss — APPROVED** [review]

File: src/main/webapp/app/exercises/shared/result/result.component.ts [hunk 1]
```diff
@@ -225,7 +226,8 @@ export class ResultComponent implements OnInit, OnChanges {
      */
     buildResultString() {
         if (this.submission && this.submission.submissionExerciseType === SubmissionExerciseType.PROGRAMMING && (this.submission as ProgrammingSubmission).buildFailed) {
-            return this.translate.instant('artemisApp.editor.buildFailed');
+            const isManualResult = this.result?.assessmentType === AssessmentType.SEMI_AUTOMATIC;
```
Comment (krusche) [inline src/main/webapp/app/exercises/shared/result/result.component.ts:229]:
```suggestion
            const isManualResult = this.result?.assessmentType !== AssessmentType.AUTOMATIC;
```

Comment (krusche) [reply src/main/webapp/app/exercises/shared/result/result.component.ts:229]: this is better to keep it compatible with older exercises

**Review by krusche — CHANGES_REQUESTED** [review]
Everything looks good, expect one small issue, see my inline comment

File: src/main/webapp/app/exercises/shared/result/result.component.ts (hunk 1 above)
Comment (fde312) [reply src/main/webapp/app/exercises/shared/result/result.component.ts:229]: Good catch! 👍

**Review by krusche — APPROVED** [review]

**Pull request merged**